state-chain/pallets/cf-ingress-egress/src/lib.rs

```diff
@@ -1098,12 +1100,17 @@ pub mod pallet {
 			pool_tier: BoostPoolTier,
 		) -> DispatchResult {
 			let booster_id = T::AccountRoleRegistry::ensure_liquidity_provider(origin)?;
+
 			ensure!(
 				T::SafeMode::get().add_boost_funds_enabled,
 				Error::<T, I>::AddBoostFundsDisabled
 			);
 			ensure!(amount > Zero::zero(), Error::<T, I>::AddBoostAmountMustBeNonZero);
 
+			// Free balances returns the unswept funds included, so if we want to be able to use
+			// those funds for boosting, we first need to sweep.
```

Comment:
nit: this comment is a little out of context - it's referring to the rpc? I think simply stating that try_debit *doesn't* implicitly sweep any open positions, and so we need to do it first, would be enough.

Reply:
updated